src/lib.rs

```diff
@@ -162,3 +208,371 @@ impl LevelIndex {
         }
     }
 }
+
+bitflags::bitflags! {
+    #[repr(transparent)]
+    pub struct ChannelTypeQualifiers: u32 {
+        const LINEAR        = (1 << 0);
+        const EXPONENT      = (1 << 1);
+        const SIGNED        = (1 << 2);
+        const FLOAT         = (1 << 3);
+        const NONE          = 0;
+        const UNINITIALIZED = 0xFFFF;
+    }
+}
+
+bitflags::bitflags! {
+    #[derive(Default)]
+    #[repr(transparent)]
+    pub struct DataFormatFlags: u32 {
```

Comment:
If this bitfield is subject to future extension, bitflags may not be a good choice for it, as some if its operators (e.g. `all`) don't make sense if there may be valid bits unknown to the code.

Reply:
I can implement something manually if you prefer.

Reply:
I'd lean towards that, maybe with a helper macro for the boilerplate. We made a similar mistake in ash and rolled it back in the end.

Reply:
Hmm, I'm basically going to have to reimplement the bitflags crate. I don't want to just copy and paste it into this crate just to remove the `all()` trait method, nor do I want to reimplement its functionality. Plus I'm not yet fluent with macros which means it will be pretty slow going for me.

Reply:
This isn't super important to me, just a minor nit. @cwfitzgerald, thoughts?

Reply:
I think leaving it as bitflags is fine, all isn't a query that is particularly relevant to this flag set, so it shouldn't be a big footgun.